### ISSUE TITLE
Trying to remove default reply-to address contains some bugs an UX problems

### ISSUE THREAD

**Issue by amazingphilippe** [cds-snc/notification-planning#684]
## Describe the bug 

Users can't remove all reply-to addresses once one is created. 
If you have one or multiple reply-to addresses, you can't delete the last final one. Once you set a reply-to, it will always keep at least one. You can't return to the initial state when you create a service of having no reply-to address

If this is expected behaviour, there are 3 bugs to fix:

1. Let the user know adding a reply-to address is irreversible. And let the user know they can't remove the default reply-to before they get to the dead-end page (see steps below)
2. It is possible to try and delete a default reply-to address by triggering field validation
3. There is no where to go once you attempt deleting a default reply-to address. 

## To Reproduce

1. Go to Settings page
2. Click on Configure reply-to address
3. Click on Change on the default reply-to address
4. Empty the input field
5. Get validation error
4. See "delete" link show up
7. Click on delete
8. Click to confirm deletion
9. See dead-end screen

## Expected behavior

Know why a default address can't be removed, before I attempt to do it anyway
Have somewhere to go when I attempt to delete a default reply-to address

## Impact

Users get stuck, we get support tickets?

If applicable

Impact on Notify users: Access to a dusty corner of Notify. Bad UX.

Impact on Recipients: None

Impact on Notify team: Support tickets?

## Screenshots

Visual outline of these steps: https://www.figma.com/file/eJKkICL6DcWWyhUEy3s9z0/?node-id=1327%3A5534

## Additional context

Do we want to let users remove the reply-to? Can we remove a default reply-to? Should we?

**Comment by yaelberger-commits** [comment]
@amazingphilippe I think this was a duplicate of the bug we just fixed #813

**Comment by smcmurtry** [comment]
Here is the relevant PR on the Gov UK Notify project where they implemented this: https://github.com/alphagov/notifications-api/pull/1847

It doesn't exactly say why you can't delete the default reply to address. I'm inclined to think you should be able to. Or at least "archive" it if we need to keep a record since it will probably be referenced from other tables.

**Comment by smcmurtry** [comment]
In the `Notifications` DB tables, there is actually no id that links to the `service_email_reply_to` table. The `Notifications` table just the text for the relevant address in the `reply_to_address` column, like `stephen.mcmurtry+reply-to-4@cds-snc.ca`. I do not yet see a technical reason why we don't let the user delete/archive all reply to addresses.

**Comment by smcmurtry** [comment]
If we allow users to delete the default reply-to address, they can then be left with multiple other reply-to addresses but none of them are default. This then causes weirdness in the UI, where the Settings page says there is no reply to address set, but when you click to set one there are already some set that aren't default.

So, if we allow the user to delete the default, we need another form or some logic to make one of the remaining addresses the default, if one exists.

**Comment by andrewleith** [comment]
API and Admin will both need some minor changes. @smcmurtry working on this today

**Comment by amazingphilippe** [comment]
Would it be simpler to not allow users to delete the default reply-to? Having one is recommended, but I understand there is no value to it if the reply-to inbox is not accessible to the service owners. 

I think we can keep the existing limitations. Just do a better job when handling form validation, or the display of the "delete" button on that template. 
https://images.zenhubusercontent.com/409232714/1d2e66ec-2a1a-463c-a76f-f9c4b3a4df8c/enregistrement_d_e_cran__le_2022_11_16_a__09_32_01.mov

**Comment by smcmurtry** [comment]
Met with @amazingphilippe this morning, we came up with a plan to fix this:
- there is no reason (that we know of) not to let the user delete the default reply-to address and return to the initial state, so
- we will let the user delete the default reply-to address
- we will display the delete button and change the api to not return an error for this (I tested this by manually performing this action in the DB and it seems to work fine)
- if the user deletes the default address, and there are other reply-to addresses, one of these should be assigned to be default. We can inform the user of this by changing the warning message to something like this:

![Screen Shot 2022-11-16 at 11.53.30 AM.png](https://images.zenhubusercontent.com/63176b6f863fa9b22e9eee20/f9a2c1ee-476b-487c-85b8-ddb29c4a152c)

**Comment by andrewleith** [comment]
- Two PRs (one for API, one for Admin)
- Waiting on final content to do review, @YedidaZalik is hopeful it could be done today
  - its ok if this has to sit waiting on content

**Comment by YedidaZalik** [comment]
EN and FR content below

**EN:**

You’re about to delete your default reply-to address. The new default will be the next email on your list of reply-to addresses: ‘[stephen.mcmurtry+replyto-7@cds-snc.ca](mailto:stephen.mcmurtry+replyto-7@cds-snc.ca)’

Delete

**FR:**

Vous êtes sur le point de supprimer l’adresse de réponse par défaut. ‘[stephen.mcmurtry+replyto-7@cds-snc.ca](mailto:stephen.mcmurtry+replyto-7@cds-snc.ca)’ deviendra votre nouvelle adresse de réponse par défaut car elle figure sur votre liste d’adresses de réponse.

Supprimer

**Comment by smcmurtry** [comment]
Reviewed by @amazingphilippe and @whabanks. Now merged and deployed in staging.

**Comment by andrewleith** [comment]
@sastels will QA in staging

**Comment by sastels** [comment]
tested adding / deleting and sending myself an email at each stag to ensure that the reply to is what it claims to be :+1:
